### PR TITLE
Add docs/concepts/architecture/controller.md in portuguese

### PR DESCRIPTION
**Pull request by joao-luna-98**
Translation of docs/concepts/architecture/controller.md to portuguese.

### CODE REVIEW

File: content/pt/docs/concepts/architecture/controller.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+Controladores que interagem com estados externos encontram o seu estado desejado
+a partir do *API server*, e então comunicam diretamente com o sistema externo para
+trazer o *estado atual* mais próximo do desejado.
+
+(Existe um controlador que escala horizontalmente nós no seu *cluster*.
+Veja [Escalamento automático do cluster](/docs/tasks/administer-cluster/cluster-management/#cluster-autoscaling))
+
+## Estado desejado versus atual {#desired-vs-current}
+
+Kubernetes leva uma visão *cloud-native* de sistemas e é capaz de manipular
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
Kubernetes tem uma visão *cloud-native* de sistemas e é capaz de manipular
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+mudanças constantes.
+
+O seu *cluster* pode mudar em qualquer altura à medida que o trabalho acontece e
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
O seu *cluster* pode mudar em qualquer momento à medida que as ações acontecem e
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+os ciclos de controlo corrigem falhas automaticamente. Isto significa que,
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
os ciclos de controle corrigem falhas automaticamente. Isto significa que,
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+potencialmente, o seu *cluster* nunca atinge um estado estável.
+
+Enquanto os controladores no seu *cluster* estiverem a correr e forem capazes de
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
Enquanto os controladores no seu *cluster* estiverem rodando e forem capazes de
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+fazer alterações úteis, não importa se o estado é estável ou se é instável.
+
+## Desenho
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]: I would keep this as Design
```suggestion
## Design
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+Como um princípio do seu desenho, o Kubernetes usa muitos controladores onde cada
+um gere um aspeto particular do estado do *cluster*. Maioritariamente, um particular
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
um gerencia um aspecto particular do estado do *cluster*. Comumente, um particular
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+ciclo de controlo (controlador) usa uma espécie de recurso como o seu *estado desejado*,
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
ciclo de controle (controlador) usa uma espécie de recurso como o seu *estado desejado*,
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+e tem uma espécie diferente de recurso que o mesmo gere para garantir que esse *estado desejado*
+é cumprido.
+
+É útil que haja controladores simples em vez de um conjunto monolítico de ciclos de controlo
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
É útil que haja controladores simples em vez de um conjunto monolítico de ciclos de controle
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+Atrás das cortinas, os controladores do Kubernetes garantem que eles apenas tomam
+atenção aos recursos ligados aos seus recursos controladores.
+
+Por exemplo, você pode ter *Deployments* e *Jobs*; ambos criam *Pods*.
+O controlador de *Job* não apaga os *Pods* que o seu *Deployment* criou,
+porque existe informação ({{< glossary_tooltip term_id="label" text="labels" >}})
+que os controladores podem usar para diferenciar esses *Pods*.
+{{< /note >}}
+
+## Formas de correr controladores {#running-controllers}
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
## Formas de rodar controladores {#running-controllers}
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+O Kubernetes vem com um conjunto de controladores embutidos que correm
+dentro do {{< glossary_tooltip term_id="kube-controller-manager" >}}.
+Estes controladores embutidos providenciam comportamentos centrais importantes.
+
+O controlador *Deployment* e o controlador *Job* são exemplos de controladores
+que veem como parte do próprio Kubernetes (controladores "embutidos").
+O Kubernetes deixa você correr o plano de controlo resiliente, para que se qualquer
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
O Kubernetes deixa você correr o plano de controle resiliente, para que se qualquer
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+um dos controladores embutidos falhar, outra parte do plano de controlo assume
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
um dos controladores embutidos falhar, outra parte do plano de controle assume
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+o trabalho.
+
+Pode encontrar controladores fora do plano de controlo, para extender o Kubernetes.
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
Pode encontrar controladores fora do plano de controle, para extender o Kubernetes.
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+Ou, se quiser, pode escrever um novo controlador você mesmo.
+Pode correr o seu próprio controlador como um conjunto de *Pods*,
+ou externo ao Kubernetes. O que encaixa melhor vai depender no que esse
+controlador faz em particular.
+
+{{% /capture %}}
+
+{{% capture whatsnext %}}
+* Leia mais sobre o [plano de controlo do Kubernetes](/docs/concepts/#kubernetes-control-plane)
```
Comment (femrtnz) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
* Leia mais sobre o [plano de controle do Kubernetes](/docs/concepts/#kubernetes-control-plane)
```

File: content/pt/docs/reference/_index.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
+---
+title: Reference
+approvers:
+- chenopis
+linkTitle: "Reference"
+main_menu: true
```
Comment (femrtnz) [inline content/pt/docs/reference/_index.md]: This page can be removed as it was created by #19689

File: content/pt/docs/reference/glossary/controller.md [hunk 15]
```diff
@@ -0,0 +1,30 @@
+---
+title: Controlador
+id: controller
+date: 2020-03-23
+full_link: /docs/concepts/architecture/controller/
+short_description: >
+  Um ciclo de controlo que observa o estado partilhado do cluster através do apiserver e efetua
```
Comment (femrtnz) [inline content/pt/docs/reference/glossary/controller.md]:
```suggestion
  Um ciclo de controle que observa o estado partilhado do cluster através do API Server e efetua
```

File: content/pt/docs/reference/glossary/controller.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+  mudanças tentando mover o estado atual em direção ao estado desejado.
+
+aka: 
+tags:
+- architecture
+- fundamental
+---
+No Kubernetes, controladores são ciclos de controlo que observam o estado do seu
```
Comment (femrtnz) [inline content/pt/docs/reference/glossary/controller.md]:
```suggestion
No Kubernetes, controladores são ciclos de controle que observam o estado do seu
```

File: content/pt/docs/reference/glossary/controller.md [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+mudanças onde necessário.
+Cada controlador tenta mover o estado atual do cluster mais perto do estado desejado.
+
+<!--more-->
+
+Controladores observam o estado partilhado do cluster através do
+{{< glossary_tooltip text="apiserver" term_id="kube-apiserver" >}} (parte do
+{{< glossary_tooltip term_id="control-plane" >}}).
+
+Alguns controladores tambem correm dentro do plano de controlo, fornecendo ciclos
```
Comment (femrtnz) [inline content/pt/docs/reference/glossary/controller.md]:
```suggestion
Alguns controladores também correm dentro do plano de controlo, fornecendo ciclos
```

File: content/pt/docs/reference/glossary/controller.md [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+de controlo que são centrais às operações do Kubernetes. Por exemplo: o controlador
```
Comment (femrtnz) [inline content/pt/docs/reference/glossary/controller.md]:
```suggestion
de controle que são centrais às operações do Kubernetes. Por exemplo: o controlador
```

**Comment by devlware** [comment]
Hey, I'll try to review it later tonight.

File: content/pt/docs/concepts/architecture/controller.md [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+Em robótica e automação um _control loop_, ou em português _ciclo de controle_, é
+um ciclo não terminado que regula o estado de um sistema.
+
+Um exemplo de ciclo de controlo é um termostato de uma sala.
+
+Quando você define a temperatura, isso indica ao termostato
+sobre o seu *estado desejado*. A temperatura ambiente real é o
+*estado atual*. O termostato atua de forma a trazer o estado atual
+mais perto do estado desejado, ligando ou desligando o equipamento.
+
```
Comment (devlware) [inline content/pt/docs/concepts/architecture/controller.md]: Hey, looks like there is one missing phrase: "In Kubernetes, controllers are control loops that watch the state of your cluster, then make or request changes where needed. Each controller tries to move the current cluster state closer to the desired state." Can you check your translation againt the last revision of the docs?

File: content/pt/docs/concepts/architecture/controller.md [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+content_template: templates/concept
+weight: 30
+---
+
+{{% capture overview %}}
+
+Em robótica e automação um _control loop_, ou em português _ciclo de controle_, é
+um ciclo não terminado que regula o estado de um sistema.
+
+Um exemplo de ciclo de controlo é um termostato de uma sala.
```
Comment (devlware) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
Um exemplo de ciclo de controle é um termostato de uma sala.
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+mais perto do estado desejado, ligando ou desligando o equipamento.
+
+{{< glossary_definition term_id="controller" length="short">}}
+
+{{% /capture %}}
+
+
+{{% capture body %}}
+
+## Padrão de controlo (Controller pattern)
```
Comment (devlware) [inline content/pt/docs/concepts/architecture/controller.md]: I'm not sure that *Controller pattern* translates to *Padrão de controlo*. I think it should something related to Controlador. Does it make sense?

File: content/pt/docs/concepts/architecture/controller.md [hunk 22]
```diff
@@ -0,0 +1,162 @@
...
+
+Um controlador rastreia pelo menos um tipo de recurso Kubernetes.
+Estes [objetos](/docs/concepts/overview/working-with-objects/kubernetes-objects/#kubernetes-objects)
+têm um campo *spec* (especificação) que representa o *estado desejado*.
```
Comment (devlware) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
têm um campo *spec* que representa o estado desejado.
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 23]
```diff
@@ -0,0 +1,162 @@
...
+Os controladores para esse recurso são responsáveis por trazer o *estado atual*
```
Comment (devlware) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
O(s) controlador(es) para aquele recurso são responsáveis por trazer o *estado atual*
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 24]
```diff
@@ -0,0 +1,162 @@
...
+mais perto do *estado desejado*.
+
+O controlador pode executar uma ação ele próprio, ou,
+o que é mais comum, enviar uma mensagem para o Kubernetes
```
Comment (devlware) [inline content/pt/docs/concepts/architecture/controller.md]:
```suggestion
o que é mais comum, no Kubernetes, o controlador envia uma mensagem para o {{< glossary_tooltip text="API server" term_id="kube-apiserver" >}} (servidor de API) que tem benefícios.
```

File: content/pt/docs/concepts/architecture/controller.md [hunk 25]
```diff
@@ -0,0 +1,162 @@
...
+{{< glossary_tooltip text="API server" term_id="kube-apiserver" >}} que tem
+efeitos colaterais úteis. Você vai ver exemplos disto abaixo.
+
+{{< comment >}}
```
Comment (devlware) [inline content/pt/docs/concepts/architecture/controller.md]: I believe this comment was removed on the last version, can you please confirm? Tks.

File: content/pt/docs/concepts/architecture/controller.md (hunk 19 above)
Comment (joao-luna-98) [reply content/pt/docs/concepts/architecture/controller.md]: That's the glossary entry for the controller term. It is translated in the references.

File: content/pt/docs/concepts/architecture/controller.md (hunk 21 above)
Comment (joao-luna-98) [reply content/pt/docs/concepts/architecture/controller.md]: Totally, my mistake.

File: content/pt/docs/reference/glossary/controller.md [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+mudanças onde necessário.
+Cada controlador tenta mover o estado atual do cluster mais perto do estado desejado.
+
+<!--more-->
+
+Controladores observam o estado partilhado do cluster através do
+{{< glossary_tooltip text="apiserver" term_id="kube-apiserver" >}} (parte do
+{{< glossary_tooltip term_id="control-plane" >}}).
+
+Alguns controladores também correm dentro do plano de controlo, fornecendo ciclos
```
Comment (jhonmike) [inline content/pt/docs/reference/glossary/controller.md:26]:
```suggestion
Alguns controladores também correm dentro do plano de controle, fornecendo ciclos
```

**Pull request merged**